scipy/optimize/optimize.py

```diff
@@ -2693,6 +2709,9 @@ def _endprint(x, flag, fval, maxfun, xtol, disp):
         if disp:
             print("\nMaximum number of function evaluations exceeded --- "
                   "increase maxfun argument.\n")
+    if flag == 2:
+        if disp:
```

Comment:
same here

Reply:
The code path is only entered for disp=True, and there are no such tests overall.